vite.config.ts

```diff
@@ -0,0 +1,25 @@
+import { defineConfig } from "vite";
+import UnpluginIsolatedDecl from 'unplugin-isolated-decl/vite'
+
+export default defineConfig({
+  build: {
+    lib: { entry: "src/index.ts",},
+    outDir: "dist",
+    emptyOutDir: true,
+    minify: false,
+    sourcemap: true,
+    rollupOptions: {
+      external: ["fastify", "zod/v4", "@fastify/swagger", "@fastify/error"],
```

Comment:
Instead of hardcoding the dependencies we could use 
https://www.npmjs.com/package/vite-plugin-externalize-deps.

Let me know if you want this change!